nucliadb/src/nucliadb/search/api/v1/augment.py

```diff
@@ -82,78 +98,202 @@ async def _augment_endpoint(
+                FieldAugment(
+                    given=item.resources.given,  # type: ignore[arg-type]
+                    select=field_select,  # type: ignore[arg-type]
+                    filter=item.resources.fields.filters,
                 )
             )
 
-        ops = [
-            augment_resources_task,
-            augment_paragraphs_task,
-        ]
+    if item.fields is not None:
+        select: list[FieldProp] = []
+        if item.fields.text:
+            select.append(FieldText())
+        if item.fields.entities:
+            select.append(FieldEntities())
+        if item.fields.classification_labels:
+            select.append(FieldClassificationLabels())
+
+        augmentations.append(
+            FieldAugment(
+                given=[FieldId.from_string(id) for id in item.fields.given],
+                select=select,
+            )
+        )
+
+    if item.paragraphs is not None:
+        paragraphs_to_augment, paragraph_selector = parse_paragraph_augment(item.paragraphs)
+        augmentations.append(
+            ParagraphAugment(
+                given=paragraphs_to_augment,
+                select=paragraph_selector,
+            )
+        )
+
+    if len(augmentations) == 0:
+        return AugmentResponse(
+            resources={},
+            fields={},
+            paragraphs={},
+        )
+
+    with request_caches():
+        max_ops = asyncio.Semaphore(50)
 
-        resources: dict[str, nucliadb_models.resource.Resource | None]
-        paragraphs: dict[ParagraphId, nucliadb.search.augmentor.models.AugmentedParagraph | None]
-        resources, paragraphs = await asyncio.gather(*ops)  # type: ignore[assignment]
+        augmented = await augmentor.augment(
+            kbid,
+            augmentations,
+            concurrency_control=max_ops,
+        )
 
         augmented_resources = {}
-        for rid, resource in resources.items():
-            if resource is None:
+        for rid, resource_deep in augmented.resources_deep.items():
+            if resource_deep is None:
                 continue
             augmented_resource = AugmentedResource(id=rid)
-            augmented_resource.updated_from(resource)
+            augmented_resource.updated_from(resource_deep)
             augmented_resources[rid] = augmented_resource
 
+        for rid, resource in augmented.resources.items():
+            if resource is None:
+                continue
+            augmented_resource = augmented_resources.setdefault(rid, AugmentedResource(id=rid))
+            augmented_resource.title = augmented_resource.title or resource.title
+            augmented_resource.summary = augmented_resource.summary or resource.summary
+            if resource.classification_labels is not None:
+                augmented_resource.classification_labels = {
+                    labelset: list(labels) for labelset, labels in resource.classification_labels.items()
+                }
+            # TODO: more properties
```

Comment:
Are we missing something needed for the ask decoupling? Otherwise, remove the todo?